lib/router/router.js

```diff
@@ -63,6 +63,13 @@ function getTransitionByIntent(intent, isIntermediate) {
 
   // Abort and usurp any previously active transition.
   if (this.activeTransition) {
+    // if you're doing multiple redirects, need the new transition to know if it
+    // is actually part of the first transition or not. Any further redirects
+    // in the initial transition also need to know if they are part of the
+    // initial transition
+    newTransition.isCausedByInitialTransition = this.activeTransition.isCausedByInitialTransition|| this.activeTransition.sequence === 0;
+    newTransition.isCausedByAbortingTransition = true;
```

Comment:
Can we set these in the Transition constructor? At the very least, we should set them to `undefined` in the constructor to avoid mutating the shape of the transition object.